dockerfiles/Dockerfile

```diff
@@ -1,15 +1,25 @@
-FROM quay.io/app-sre/qontract-reconcile-base:0.5.1
+FROM quay.io/app-sre/qontract-reconcile-builder:0.1.0 as build-image
 
-WORKDIR /reconcile
+WORKDIR /work
 
-COPY e2e_tests e2e_tests
-COPY reconcile reconcile
-COPY tools tools
-COPY setup.py .
+COPY e2e_tests \
+     reconcile \
+     tools \
+     setup.py \
+     dockerfiles/hack/run-integration.py \
+     /work/
+
+RUN python3 -m pip install --upgrade pip setuptools wheel && \
+    python3 -m pip wheel . --wheel-dir /work/wheels
+
+
+FROM quay.io/app-sre/qontract-reconcile-base:0.6.0
+
+RUN --mount=type=cache,target=/tmp/work/,from=build-image,source=/work \
+    microdnf upgrade -y && \
```

Comment:
I like the idea of the upgrade every time we build. I would try to make sure that someone else reads this in case there's a strong disagreement :)